ggml/src/ggml-sycl/presets.hpp

```diff
@@ -16,7 +16,11 @@
 #define GGML_SYCL_MAX_STREAMS       8
 #define GGML_SYCL_MAX_BUFFERS       256
 
+#ifdef GGML_SYCL_NV_ARCH
 #define WARP_SIZE 32
+#else
+#define WARP_SIZE 16
+#endif
```

Comment:
At present GGML_SYCL_NV_ARCH is never defined so I had to patch this to be able to build. I think it would be useful to hook this up to `ggml/src/CMakeLists.txt`, where we have e.g.:

```CMake
        if (GGML_SYCL_TARGET STREQUAL "INTEL")
           #...
        elseif (GGML_SYCL_TARGET STREQUAL "NVIDIA")
           #...
        endif()
```

Secondly, I think many (most?) Intel platforms support both 32 and 16 for sg_size, so maybe you want to make it a CMake switch? For Codeplay's perspective, that is less important since we'll always use 32. Just a suggestion

Reply:
It seems that GGML_SYCL_NV_ARCH was removed by my previous rebase. I will add it back later.
I think 32 is just for compatibility. For the current Intel archs, 32 has no performance advantages but bugs. Using 16 only is okay for now.

Reply:
add back the macro definition.